### PR TITLE
Add function Math::lerp, Vector3::lerp and Vector2::lerp

### PR DESCRIPTION
**Pull request by fuyutsuki**
Theory: <https://en.wikipedia.org/wiki/Linear_interpolation>

### CODE REVIEW

File: src/Vector2.php [hunk 1]
```diff
@@ -144,6 +144,13 @@ public function lengthSquared() : float{
 		return $this->x * $this->x + $this->y * $this->y;
 	}
 
+	public function lerp(Vector2 $start, Vector2 $end, float $param): Vector2 {
```
Comment (dktapps) [inline src/Vector2.php]: `$param` is too generic. It gives no indication what the parameter is for.

File: src/Vector3.php [hunk 2]
```diff
@@ -293,6 +293,13 @@ public function lengthSquared() : float{
 		return $this->x * $this->x + $this->y * $this->y + $this->z * $this->z;
 	}
 
+	public function lerp(Vector3 $start, Vector3 $end, float $param): Vector3 {
```
Comment (dktapps) [inline src/Vector3.php]: Same again here. Too generic.

**Comment by fuyutsuki** [comment]
> It appears that the functions ought to be static.

I decided that it is not necessary to make static because `Vector3` and `Vector2` functions are not static, but should it still be static?

**Comment by dktapps** [comment]
In their current form they make no sense. `(new Vector3())->lerp(new Vector3(0, 0, 0), new Vector3(1, 1, 1), 0.5)`

**Comment by fuyutsuki** [comment]
I made some strange mistake lol

**Comment by dktapps** [comment]
please refer to the discussion on #6

**Comment by dktapps** [comment]
oops, I meant to refer to #5 ... the point being that a lot of these functions don't make sense as-is.

**Comment by dktapps** [comment]
yes

**Comment by fuyutsuki** [comment]
Is there anything else I should change?

File: src/Math.php [hunk 3]
```diff
@@ -75,4 +75,20 @@ public static function solveQuadratic(float $a, float $b, float $c) : array{
...
+	/**
+	 * Returns a linearly interpolated value at a percentage specified by $percent in the range from $start to $end
+	 *
+	 * @param number $start
+	 * @param number $end
+	 * @param float $percent
+	 *
+	 * @return float
+	 */
+	public static function lerp($start, $end, float $percent) : float{
```
Comment (dktapps) [inline src/Math.php]: pls use `float` for these parameters

**Comment by dktapps** [comment]
@pmmp/developers can I get a second pair of eyes on this?

File: tests/phpunit/LerpTest.php [hunk 4]
```diff
@@ -27,8 +27,8 @@ class LerpTest extends TestCase{
 
 	public function someLengthProvider() : \Generator{
 		yield [
-			1,
-			10,
+			(float)1,
```
Comment (dktapps) [inline tests/phpunit/LerpTest.php]: this is unnecessary

**Comment by fuyutsuki** [comment]
I will recreate this when issue #3 and #4 is solved

**Pull request closed without merging**